pkg/controller/gitlab/buckets_test.go

```diff
@@ -65,11 +65,31 @@ func getBucketClaimType(name string) string {
 	return bucketClaimKind + "-" + name
 }
 
+func assertBucketObject(t *testing.T, testName string, obj runtime.Object) *xpstoragev1alpha1.Bucket {
+	bucket, ok := obj.(*xpstoragev1alpha1.Bucket)
+	if !ok {
+		t.Errorf("%s unexpected type: %T", testName, obj)
+		return nil
+	}
+	return bucket
+}
+
 var _ resourceReconciler = &bucketReconciler{}
 
 func Test_bucketReconciler_reconcile(t *testing.T) {
 	ctx := context.TODO()
+	testCaseName := "bucketReconciler.reconcile()"
 	testError := errors.New("test-error")
+
+	assertBucketName := func(obj runtime.Object) *xpstoragev1alpha1.Bucket {
+		b := assertBucketObject(t, testCaseName, obj)
+		if diff := cmp.Diff(b.Spec.Name,
+			testName+"-"+xpstoragev1alpha1.BucketKind+"-"+testBucket+bucketNameDelimiter+"%s"); diff != "" {
+			t.Errorf("%s unexpected name: %s", testCaseName, diff)
+		}
```

Comment:
This reads a little oddly to me given that the condition and result of the condition end up at the same level of indentation. I'd prefer assigning the wanted name to a variable named `want` before the `if diff` line.

I'd also prefer we order the `cmp.Diff()` args as (`cmp.Diff(want, got)` rather than `cmp.Diff(got, want)` per my (perhaps subjective) reasoning in https://github.com/crossplaneio/crossplane/issues/452, but I understand that the got, want style is consistent with the rest of this file.